.github/workflows/test.yml

```
@@ -0,0 +1,60 @@
+on:
+  pull_request:
+    branches:
+      - main
+      - master
+  push:
+    branches:
+      - main
+      - master
+  release:
+    types: [published]
+
+env:
+  GOFLAGS: "-mod=vendor"
+  GO111MODULE: on
+
+jobs:
+  test:
+    name: Test and Publish
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
```

The [goreleaser action docs](https://github.com/goreleaser/goreleaser-action?tab=readme-ov-file#workflow) recommend checking out with a fetch depth of zero.

```suggestion
        uses: actions/checkout@v4
        with:
          fetch-depth: 0
```

---

Thanks!